encoding/thrift/inbound.go

```diff
@@ -209,3 +202,28 @@ func closeReader(r io.Reader) error {
 
 	return closer.Close()
 }
+
+// getBody avoids redundant copy if the transport request is already
+// io.ReaderAt compatible. If not, it creates buffer using bufferpool and returns
+// deferFn to delay bufferpool release
+// This is mainly done as tchannel transport handler decodes the body already and
+// sets it to io.ReaderAt compatible instance
+func getBody(treq *transport.Request) (body io.ReaderAt, deferFn func(), err error) {
```

Comment:
I would rename `deferFn` to `release` -> developer can use the release function the way they want, it is not tied to be used in a defer statement. It also makes more sense since it will release the io